app/controllers/news_items_controller.rb

```diff
@@ -1,9 +1,17 @@
 class NewsItemsController < ApplicationController
   def index
-    @news_items = NewsItem.published.to_a
+    @news_items = news_item_scope.to_a
   end
 
   def show
-    @news_item = NewsItem.published.find(params[:id])
+    @news_item = news_item_scope.where(slug: params[:slug]).first ||
+                 news_item_scope.where(id: params[:id]).first ||
+                 raise(ActiveRecord::RecordNotFound)
```

Comment:
@benichu Could we extract as a method and pushed into the model? e.g.

```ruby
NewsItem.fetch_published!(params[:id], params[:slug])
```

This allow to document the method, as I'm unsure why this is required.